src/renderer/viz/expressions/Ramp.js

```diff
@@ -0,0 +1,265 @@
+import BaseExpression from './base';
+import { implicitCast, checkExpression, checkMaxArguments } from './utils';
+
+import RampImage from './RampImage';
+import RampGeneric from './RampGeneric';
+import { DEFAULT_OTHERS } from './constants';
+
+/**
+* Create a ramp: a mapping between an input (a numeric or categorical expression) and an output (a color palette or a numeric palette, to create bubble maps)
+*
+* Categories to colors
+* Categorical expressions can be used as the input for `ramp` in combination with color palettes. If the number of categories exceeds the number of available colors in the palette new colors will be generated by
+* using CieLAB interpolation.
+*
+* Categories to numeric
+* Categorical expression can be used as the input for `ramp` in combination with numeric palettes. If the number of input categories doesn't match the number of numbers in the numeric palette, linear interpolation will be used.
+*
+* Numeric expressions to colors
+* Numeric expressions can be used as the input for `ramp` in combination with color palettes. Colors will be generated by using CieLAB interpolation.
+*
+* Numeric expressions to numeric
+* Numeric expressions can be used as the input for `ramp` in combination with numeric palettes. Linear interpolation will be used to generate intermediate output values.
+*
+* @param {Number|Category} input - The input expression to give a color
+* @param {Palette|Color[]|Number[]} palette - The color palette that is going to be used
+* @return {Number|Color}
+*
+* @example <caption>Mapping categories to colors and numbers</caption>
+* const s = carto.expressions;
+* const viz = new carto.Viz({
+*   width: s.ramp(s.buckets(s.prop('dn'), [20, 50, 120]), [1, 4, 8])
+*   color: s.ramp(s.buckets(s.prop('dn'), [20, 50, 120]), s.palettes.PRISM)
+* });
+*
+* @example <caption>Mapping categories to colors and numbers (String)</caption>
+* const viz = new carto.Viz(`
+*   width: ramp(buckets($dn, [20, 50, 120]), [1, 10,4])
+*   color: ramp(buckets($dn, [20, 50, 120]), prism)
+* `);
+*
+*
+* @example <caption>Mapping numeric expressions to colors and numbers</caption>
+* const s = carto.expressions;
+* const viz = new carto.Viz({
+*   width: s.ramp(s.linear(s.prop('dn'), 40, 100), [1, 8])
+*   color: s.ramp(s.linear(s.prop('dn'), 40, 100), s.palettes.PRISM)
+* });
+*
+* @example <caption>Mapping numeric expressions to colors and numbers (String)</caption>
+* const viz = new carto.Viz(`
+*   width: ramp(linear($dn, 40, 100), [1, 10,4])
+*   color: ramp(linear($dn, 40, 100), prism)
+* `);
+*
+* @memberof carto.expressions
+* @name ramp
+* @function
+* @api
+*/
+export default class Ramp extends BaseExpression {
+    constructor (input, palette, others = DEFAULT_OTHERS) {
```

Comment:
`DEFAULT_OTHERS` shouldn't be in another file since this is local to `Ramp`

Reply:
but it is shared between all Ramp classes (we have 3 now)